tests/kernel/nfs_server.pm

```diff
@@ -0,0 +1,101 @@
+# SUSE's openQA tests
+#
+# Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: NFS server
+#    This module provisions the NFS server and then runs some
+#    basic sanity tests
+# Maintainer: Kernel QE <kernel-qa@suse.de>
+
+use Mojo::Base "opensusebasetest";
+use testapi;
+use serial_terminal "select_serial_terminal";
+use lockapi;
+use utils;
+use Utils::Logging "export_logs_basic";
+
+sub run {
+    select_serial_terminal();
+
+    #TODO: configure nfs config as pleased, specifically:
+    #USE_KERNEL_NFSD_NUMBER
+    #NFS3_SERVER_SUPPORT
+    #NFS4_SUPPORT
+    #NFSV4LEASETIME
+    #TODO: introduce nfs ha setup
+
+    #TODO: all below folder creation and exports should be abstracted
+    #TODO: allow this to be configurable
+    my $nfs_mount_nfs3 = "/nfs/shared_nfs3";
+    my $nfs_mount_nfs4 = "/nfs/shared_nfs4";
+    my $nfs_permissions = "rw,sync,no_root_squash";
+
+    my $nfs_mount_nfs3_async = "/nfs/shared_nfs3_async";
+    my $nfs_mount_nfs4_async = "/nfs/shared_nfs4_async";
+    my $nfs_permissions_async = "rw,async,no_root_squash";
+
+    # provision NFS server(s) of various types
+    zypper_call("in yast2-nfs-server");
+
+    assert_script_run("mkdir -p $nfs_mount_nfs3");
+    assert_script_run("chmod 777 $nfs_mount_nfs3");
+    assert_script_run("mkdir -p $nfs_mount_nfs4");
+    assert_script_run("chmod 777 $nfs_mount_nfs4");
+
+    assert_script_run("mkdir -p $nfs_mount_nfs3_async");
+    assert_script_run("chmod 777 $nfs_mount_nfs3_async");
+    assert_script_run("mkdir -p $nfs_mount_nfs4_async");
+    assert_script_run("chmod 777 $nfs_mount_nfs4_async");
+
+    assert_script_run("echo $nfs_mount_nfs3 client-node00\\($nfs_permissions\\) >> /etc/exports");
+    assert_script_run("echo $nfs_mount_nfs4 client-node00\\($nfs_permissions\\) >> /etc/exports");
+    assert_script_run("echo $nfs_mount_nfs3_async client-node00\\($nfs_permissions_async\\) >> /etc/exports");
+    assert_script_run("echo $nfs_mount_nfs4_async client-node00\\($nfs_permissions_async\\) >> /etc/exports");
+
+    record_info("EXPORTS", script_output("cat /etc/exports"));
+
+    systemctl("enable nfs-server");
+    systemctl("start nfs-server");
+    systemctl("is-active nfs-server");
+
+    record_info("NFS config", script_output("cat /etc/sysconfig/nfs"));
+
+    #my $nfsstat = script_output("nfsstat -s");
+    record_info("NFS stat for server", script_output("nfsstat -s"));
```

Comment:
Another statement-checking tool is nfsdclnts, it depend on package python3-pyyaml or python310-pyaml, and it use to check nfs client status.

Reply:
FYI `nfsdclnts` was added in [nfs-utils-2-4-4-rc5](https://git.linux-nfs.org/?p=steved/nfs-utils.git;a=commit;h=f7e6d6b678e2313c50552ea079b85258690d49c9), first public release was in 2.5.1 => only in Tumbleweed, not in 15-SP4.